leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+                // Constant and Stochastic
+                var seg1 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: initValue, startTP: '0', stopTP: 'A'}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'R', refEvidencePair: '0000', startTP: 'A', stopTP: 'Infinity'}); 
+            } else if (funcType == 'MP') {
+                // Increase and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'I', refEvidencePair: null, startTP: '0', stopTP: 'A'}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: null, startTP: 'A', stopTP: 'Infinity'}); 
+            } else if (funcType == 'MN') {
+                // Decrease and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'D', refEvidencePair: null, startTP: '0', stopTP: 'A'}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: null, startTP: 'A', stopTP: 'Infinity'}); 
+            } else if (funcType == 'SD') {
+                // Constant and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: '0011', startTP: '0', stopTP: 'A'}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: '1100', startTP: 'A', stopTP: 'Infinity'}); 
+                analysisRequest.getUserEvaluationByID(this.get('id'), "0").evaluationValue = '0011';
+            } else if (funcType == 'DS') {
+                // Constant and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: '1100', startTP: '0', stopTP: 'A'}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: '0011', startTP: 'A', stopTP: 'Infinity'}); 
+                analysisRequest.getUserEvaluationByID(this.get('id'), "0").evaluationValue = '1100';
+            }
+            this.evolvingFunction.get('functionSegList').push(seg1, seg2);
+        }
+    }, 
+ 
+    /**
+     * Adds a new Constraint object int the global model variable,
+     * representing an absolute constraint, if requried.
+     *
+     * If funcType is RC, CR, MP, MN, SD, DS  a Constraint object
+     * representing an absolute constraint will be added. If not, this
+     * function does not do anything
+     *
+     * @param {String} funcType
+     *   ex: 'RC'
+     */
+    addAbsConst: function(funcType) {
+        if (funcType == 'RC' || funcType == 'CR' || funcType == 'MP' ||
+            funcType == 'MN' || funcType == 'SD' || funcType == 'DS') {
+            // TODO - we are deleteing model i think, so how do we access constraints now???
+            model.constraints.push(new ConstraintBBM({type: 'A', srcID: this.get('id'), srcRefTP: 'A', destID: null, destRefTP: null}));
+        }
+    }, 
+ 
+ 
+    /**
+     * Returns the absolute time for this Intention's absolute constraint at
+     * the starting epoch boundary start
+     *
+     * @param {String} source
+     *  ex. 'A'
+     */
+    getAbsConstTime: function(source) {
```

Comment:
I think this will actually be accessing the absTP param on the relevant function segment now. @amgrubb is that correct?